cmd/spire-server/cli/entry/util.go

```diff
@@ -91,6 +63,36 @@ func printEntry(e *common.RegistrationEntry) {
 	fmt.Println()
 }
 
+func parseFile(path string) ([]*types.Entry, error) {
+	dat, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	commonEntries := &common.RegistrationEntries{}
+	if err := json.Unmarshal(dat, &commonEntries); err != nil {
+		return nil, err
+	}
+
+	return api.RegistrationEntriesToProto(commonEntries.Entries)
+}
+
+func idStringToProto(id string) (*types.SPIFFEID, error) {
+	idType, err := spiffeid.FromString(id)
+	if err != nil {
+		return nil, err
+	}
+	return &types.SPIFFEID{
+		TrustDomain: idType.TrustDomain().String(),
+		Path:        idType.Path(),
+	}, nil
+}
+
+// ProtoToIDString converts types.SPIFFEID into an ID string
+func protoToIDString(id *types.SPIFFEID) string {
+	return fmt.Sprintf("spiffe://%s%s", id.TrustDomain, id.Path)
```

Comment:
can we be sure that path always contains `/`?  maybe we can parse proto to spiffeid.ID and then print strnig from there?

Reply:
The path value **should** start with a leading slash (according to the types.SPIFFEID docs). This internal function is only used for printing the entries that come from the server. I believe there is not much to do at this point if some of the entry SPIFFE IDs are somehow malformed. Probably is a good idea to print them as they come to uncover any issue.